fix: Resend invite operation on users list

RicardoE105 opened the pull request:
## Summary

Pending users were added into the store with the property `globalRole`, but when we return the users we save them in the store with the property `role` instead.

![image](https://github.com/user-attachments/assets/3206f870-3d2c-47c3-a876-50ca9273d753)


## Before

See video on ticket -> https://linear.app/n8n/issue/HELP-645/resend-invitation-link-does-not-work-anymore

## Now

![CleanShot 2024-10-22 at 16 10 25](https://github.com/user-attachments/assets/1a41705a-e7e5-4f46-85d2-478012fd0f3e)


<!--
Describe what the PR does and how to test.
Photos and videos are recommended.
-->

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
-->

https://linear.app/n8n/issue/HELP-645/resend-invitation-link-does-not-work-anymore

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

RicardoE105 committed:
fix resend invite operation

netroy reviewed (dismissed):
since `users.store` already has a test file, can we please add a test case to cover for this? that way we can avoid the same regression in the future.

RicardoE105 committed:
add test

despairblue commented on packages/editor-ui/src/stores/users.store.test.ts:
```diff
@@ -58,4 +63,29 @@ describe('users.store', () => {
+
+			inviteUsers.mockResolvedValueOnce([
+				{
+					user: { role: 'global:member', email: 'test@n8n.io', emailSent: true },
```
```suggestion
					user: { id: 'random-id', email: 'test@n8n.io', emailSent: true },
```

The API does not return the role, but the id.

despairblue commented on packages/editor-ui/src/stores/users.store.test.ts:
```diff
@@ -58,4 +63,29 @@ describe('users.store', () => {
+
+			expect(usersStore.allUsers[0]).toMatchObject(
+				expect.objectContaining({
+					role: 'global:member',
```
```suggestion
					role: 'global:admin',
```

RicardoE105 replied:
Why do we need this?/

despairblue replied:
On line 77 it says the role is `global:admin`. We can also change line 77 to `global:member`.

despairblue reviewed (changes requested):
The test wasn't preventing a regression. The API mock contained the role, but in reality this happens because the API does not return the role, right? 

Which begs the question if the store code could be simplified by changing the API?

RicardoE105 committed:
Update packages/editor-ui/src/stores/users.store.test.ts

Co-authored-by: Danny Martini <danny@n8n.io>

RicardoE105 commented:
@despairblue yes, added the role there is an overlooked. Just committed your change. With the ID and not returning the role still works.

despairblue commented:
> @despairblue yes, adding the role there is an overlooked. Just committed your change. Adding the the ID and not returning the role still works.

Would it be cleaner to add the `role` property to the API response?

despairblue committed:
return role from `POST /invitations`

This makes sure the role is in the user store in the frontend without
the FE needing to do manual remapping.

despairblue commented:
@RicardoE105 this is what I mean:

https://github.com/n8n-io/n8n/commit/a6565d3cbf869ae9dba025238e6650dbbfb485e0

Just wondering if that is cleaner, than having the FE retain data it sends to the API and merge it with the response.

RicardoE105 commented:
@despairblue gotcha now. Yet let's do that. Will include that change if this PR if you do not mind.

RicardoE105 committed:
Merge branch 'help-645-resend-invitation-link-does-not-work-anymore-alt' into help-645-resend-invitation-link-does-not-work-anymore

RicardoE105 committed:
use role from API response

RicardoE105 committed:
fix test

janober commented:
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)
